tests/integration/helpers.py

```diff
@@ -216,10 +217,45 @@ def check_tls(ip: str, port: int) -> bool:
         return False
 
 
+def consume_and_check(model_full_name: str, provider_unit_name: str, topic: str) -> None:
+    """Consumes 15 messages created by `produce_and_check_logs` function.
+
+    Args:
+        model_full_name: the full name of the model
+        provider_unit_name: the app to grab credentials from
+        topic: the desired topic to consume from
+    """
+    relation_data = get_provider_data(
+        unit_name=provider_unit_name,
+        model_full_name=model_full_name,
+        endpoint="kafka-client-admin",
+    )
+    topic = topic
+    username = relation_data.get("username", None)
+    password = relation_data.get("password", None)
+    servers = relation_data.get("endpoints", "").split(",")
+    security_protocol = "SASL_PLAINTEXT"
+
+    if not (username and password and servers):
+        raise KeyError("missing relation data from app charm")
+
+    client = KafkaClient(
+        servers=servers,
+        username=username,
+        password=password,
+        security_protocol=security_protocol,
+    )
+
+    client.subscribe_to_topic(topic_name=topic)
+    messages = [*client.messages()]
+
+    assert len(messages) == 15
```

Comment:
**niptick** Just to avoid magic number like 15, maybe we could just return the `len(messages)` outside the function and do the assert in the tests

Reply:
I don't think it really matters. There is no benefit to being able to change this number, we don't want a specific number, just _a_ number of messages. All calls to this function would change to `produce_and_check_logs(..., messages=15)`. Similarly, we just want to check that those messages exist when consuming.

Reply:
Maybe put `TEST_MESSAGE_COUNT=15` global